packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SingleImageViewer/SingleImageViewer.js

```diff
@@ -1,112 +1,132 @@
-import styled, { css } from 'styled-components'
+import { arrayOf, bool, func, number, shape, string } from 'prop-types'
+import { useEffect } from 'react'
 
-import SVGContext from '@plugins/drawingTools/shared/SVGContext'
-import InteractionLayer from '../InteractionLayer'
 import ZoomControlButton from '../ZoomControlButton'
-import locationValidator from '../../helpers/locationValidator'
 
-const PlaceholderSVG = styled.svg`
-  background: no-repeat center / cover url('https://static.zooniverse.org/www.zooniverse.org/assets/fe-project-subject-placeholder-800x600.png');
-  touch-action: pinch-zoom;
-  max-width: ${props => props.$maxWidth || '100%'};
-  ${props => props.$maxHeight && css`max-height: ${props.$maxHeight};`}
-`
-const SVGImageCanvas = styled.svg`
-  overflow: visible;
-`
+import VisXZoom from '../SVGComponents/VisXZoom'
+
+import SingleImageCanvas from './SingleImageCanvas'
+
+const DEFAULT_HANDLER = () => true
+const DEFAULT_ZOOM_CONFIG = {
+  direction: 'both',
+  maxZoom: 10,
+  minZoom: 0.1,
+  onWheelThrottleWait: 100,
+  zoomInValue: 1.2,
+  zoomOutValue: 0.8
+}
+
 function SingleImageViewer({
-  children,
-  enableInteractionLayer = false,
+  enableInteractionLayer = true,
+  enableRotation = DEFAULT_HANDLER,
   frame = 0,
-  height,
-  limitSubjectHeight = false,
-  onKeyDown = () => true,
-  rotate = 0,
-  svgMaxHeight = null,
-  title = {},
-  viewBox,
-  width,
+  imgRef,
+  invert = false,
+  move = false,
+  naturalHeight,
+  naturalWidth,
+  onKeyDown = DEFAULT_HANDLER,
+  panning = true,
+  rotation = 0,
+  setOnPan = DEFAULT_HANDLER,
+  setOnZoom = DEFAULT_HANDLER,
+  src,
+  subject,
+  title = undefined,
   zoomControlFn = null,
-  zooming = false
+  zooming = true
 }) {
-  const canvasLayer = useRef()
-  const canvas = canvasLayer.current
-  const transform = `rotate(${rotate} ${width / 2} ${height / 2})`
+  useEffect(function onMount() {
+    enableRotation()
+  }, [])
+
+  const singleImageCanvasProps = {
+    enableInteractionLayer,
+    frame,
+    imgRef,
+    invert,
+    move,
+    naturalHeight,
+    naturalWidth,
+    onKeyDown,
+    rotation,
+    src,
+    subject
+  }
 
   return (
-    <SVGContext.Provider value={{ canvas, viewBox, rotate, width, height }}>
+    <>
       {zoomControlFn && (
         <ZoomControlButton
           onClick={zoomControlFn}
           zooming={zooming}
         />
       )}
       <Box
+        align='flex-end'
         animation='fadeIn'
         overflow='hidden'
         width='100%'
-        align='flex-end'
       >
-        <PlaceholderSVG
-          focusable
-          $maxHeight={svgMaxHeight}
-          $maxWidth={limitSubjectHeight ? `${width}px` : '100%'}
-          onKeyDown={onKeyDown}
-          tabIndex={0}
-          viewBox={`0 0 ${width} ${height}`}
+        {title?.id && title?.text && (
+          <title id={title.id}>{title.text}</title>
+        )}
+        <svg
+          style={{ touchAction: move ? 'none' : 'pinch-zoom' }}
```

Comment:
I'm not 100% sure about `touch-action: none` if in `move` mode, but the UX I'm addressing with that is when in `move` mode, if I start to pinch to zoom, it often starts dragging briefly before the pinch registers and the image jumps around. With `touch-action: none` in `move` mode a user can't pinch to zoom, which might be unexpected, but the toolbar buttons will zoom, and going to `annotate` mode allows pinch-zoom, so that's the reasoning.

Reply:
Move mode on a touchscreen works much better! It's a smooth drag experience I can use the image toolbar to zoom and rotate.

I actually don't think that annotate mode need pinch-zoom, do you? For instance, when in annotate mode and I try to pinch the viewer, it draws a mark on my phone anyway. Can we use `touchAction: 'none'` for both modes?

Reply:
Yes I agree, I use the toolbar zoom buttons in annotate mode, and felt like the pinch zoom in annotate mode wasn't ideal (creates a mark like you note, and once zoomed in if I want to pan to make another mark, I have to pinch zoom out, then toggle to move mode, then pan, compared to using toolbar zoom buttons the toolbar and notably pan/zoom button stays in view).

I've updated to just `touchAction: 'none'` and we can revisit if there's feedback otherwise.